script.js

```diff
@@ -1 +1,252 @@
+    cardElement.innerText = clickedCard.display;
+    setTimeout(() => {
+    // clear the data of the first card
+      firstCard = null;
+      // make the card data disappear from user's view
+      firstCardElement.innerText = '';
+      cardElement.innerText = '';
+
+      // re-enable clicking
+      canClick = true;
+    }, 2000);
+    outputBox.innerText = 'No match! :( Try again';
+  }
+};
+
+// ------------build the board's elements-------------------------------
+const buildBoardElements = (board) => {
+  // create the element that everything will go inside of
+  const boardElement = document.createElement('div');
+
+  // give it a class for CSS purposes
+  boardElement.classList.add('board');
+
+  // use the board data structure we passed in to create the correct size board
+  for (let i = 0; i < board.length; i += 1) {
+    // make a var for just this row of cards
+    const row = board[i];
+
+    // make an element for this row of cards
+    const rowElement = document.createElement('div');
+    rowElement.classList.add('row');
+
+    // make all the squares for this row
+    for (let j = 0; j < row.length; j += 1) {
+      // create the square element
+      const square = document.createElement('div');
+
+      // set a class for CSS purposes
+      square.classList.add('square');
+
+      // set the click event
+      // eslint-disable-next-line
+            square.addEventListener('click', (event) => {
+
+        // we will want to pass in the card element so
+        // that we can change how it looks on screen, i.e.,
+        // "turn the card over"
+        if ((canClick === true) && (gameMode == START_GAME)) {
+          squareClick(event.currentTarget, i, j);
+        }
+      });
+
+      rowElement.appendChild(square);
+    }
+    boardElement.appendChild(rowElement);
+  }
+
+  return boardElement;
+};
+
+// ========Game flow================================
+const gameInit = () => {
+  // /create an outputbox
+  outputBox = document.createElement('div');
+  outputBox.classList.add('outputBox');
+  outputBox.innerText = 'Welcome! try to find two matching cards in the grid below!';
+
+  // create this special deck by getting the doubled cards and
+  // making a smaller array that is ( boardSize squared ) number of cards
+  const doubleDeck = makeDeck();
+  const deckSubset = doubleDeck.slice(0, boardSize * boardSize);
+  deck = shuffleCards(deckSubset);
+
+  // deal the cards out to the board data structure
+  for (let i = 0; i < boardSize; i += 1) {
+    board.push([]);
+    for (let j = 0; j < boardSize; j += 1) {
+      board[i].push(deck.pop());
+    }
+  }
+
+  const boardEl = buildBoardElements(board);
+
+  document.body.appendChild(outputBox);
+  document.body.appendChild(boardEl);
+};
+//= ========Start the game================================
+gameInit();
+
+setTimeout(() => {
+  gameMode = TIMES_UP;
+  outputBox.innerText = ' Time\'s up! (3 mins) Click refresh the page and try again ';
+}, gameDurationis3Min);
```

Comment:
This logic should not be done at the highest level of the code. It should be done as part of game initialisation in `gameInit`, or when the user first starts the game by interacting with it somehow.

Having this logic here is out of place because it isn't grouped with any of the other logical segments of the game. If it should happen at the game initialisation, it should be in `gameInit` function.